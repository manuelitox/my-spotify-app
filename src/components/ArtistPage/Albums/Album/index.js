import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import BodyClassName from 'react-body-classname'

import PopUpAlbum from './PopUp/'
import StylesAlbum from './styles'
import PlaceholderAlbum from 'svgs/placeholderAlbum.svg'

class Album extends Component {
  state = { isOpen: false, id: null }

  selectToAlbumId = id => this.setState({ id })

  togglePopUp = id => {
    const albumsListPopup = document.getElementById('albums-list-popup')
    albumsListPopup && albumsListPopup.scrollTo(0, 0);
    this.setState({ isOpen: !this.state.isOpen })
    this.selectToAlbumId(id)
  }

  render () {
    const { isOpen } = this.state
    const { id, cover, title, releaseDate, numberTracks } = this.props
    return (
      <Fragment>
        { isOpen && <BodyClassName className="disable-albums-list-popup" /> }
        <StylesAlbum onClick={ this.togglePopUp.bind(this, id) }>
          <img 
            src={ cover ? cover : PlaceholderAlbum } 
            alt={ title }
            width="60"
            height="60" />
          <div>
            <h3>{ title }</h3>
            <p>Total tracks: { numberTracks }</p>
            <p>Release date: { releaseDate }</p>
          </div>
        </StylesAlbum>      
        <PopUpAlbum 
          cover={ cover }
          albumId={ this.state.id }
          isOpen={ this.state.isOpen } 
          togglePopUp={ this.togglePopUp }
          title={ title }
          releaseDate={ releaseDate }
          numberTracks={ numberTracks }/>
      </Fragment>
    )
  }
}

Album.propTypes = {
  id: PropTypes.string,
  cover: PropTypes.string,
  title: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  numberTracks: PropTypes.number.isRequired
}

export default Album