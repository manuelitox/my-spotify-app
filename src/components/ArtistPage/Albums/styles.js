import styled from 'styled-components'

const StyledWrapperAlbums = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 30px;
  @media only screen and (max-width: ${ props => props.theme.size.small }) {
    grid-template-columns: 1fr
  }
`

export default StyledWrapperAlbums