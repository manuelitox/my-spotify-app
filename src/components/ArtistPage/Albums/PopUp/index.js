import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

import theme from 'config/theme'
import PopUp from 'components/generic/PopUp/container'
import Title from 'components/ArtistPage/Title/'
import Album from 'components/ArtistPage/Albums/Album/'
import StyledWrapperAlbums from 'components/ArtistPage/Albums/styles'
import getImage from 'lib/get-image'

const PopUpAlbums = ({ isOpen, togglePopUp, albums, className }) => (
  <PopUp isOpen={ isOpen } togglePopUp={ togglePopUp } className={ className }>
    <Fragment>
      <Title color={ theme.blueDark } insidePopUp={ true }>Albums</Title>
      <StyledWrapperAlbums inPopup>
        { albums.map( (album, index) => (
          <Album key={ index } id={ album.id } title={ album.name } cover={ getImage(album.images) } numberTracks={ album.total_tracks } releaseDate={ album.release_date } />
        )) }
      </StyledWrapperAlbums>
    </Fragment>
  </PopUp>      
)

PopUpAlbums.propTypes = {
  isOpen: PropTypes.bool,
  className: PropTypes.string,
  togglePopUp: PropTypes.func.isRequired,
  albums: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      total_tracks: PropTypes.number,
      release_date: PropTypes.string
    })
  )
}

export default PopUpAlbums