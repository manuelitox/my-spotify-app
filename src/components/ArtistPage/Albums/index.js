import React from 'react'

import Title from 'components/ArtistPage/Title/'
import Album from './Album/'
import Button from 'components/generic/Button/'
import StyledWrapperAlbums from './styles'

const Albums = () => (
  <div>
    <Title>Albums</Title>
    <StyledWrapperAlbums>
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
      <Album title="Hallelujah" numberTracks={ 12 } releaseDate="2016-06-17" />
    </StyledWrapperAlbums>
    <Button onClick={ () => console.log('open albums list popup') }>view more albums</Button>
  </div>
)

export default Albums