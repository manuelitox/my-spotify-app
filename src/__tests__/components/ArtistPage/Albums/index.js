import React from 'react'
import { shallow } from 'enzyme'

import Albums from 'components/ArtistPage/Albums/'

describe('Component | Albums', () => {
  let component 
  beforeEach( () => {
    component = shallow(
      <Albums />
    )
  })
  it('should render a component', () => {
    expect(component).toMatchSnapshot()
  })
})