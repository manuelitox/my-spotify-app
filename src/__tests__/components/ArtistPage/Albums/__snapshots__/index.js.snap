// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | Albums should render a component 1`] = `
<div>
  <Title>
    Albums
  </Title>
  <styled.div>
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
    <Album
      numberTracks={12}
      releaseDate="2016-06-17"
      title="Hallelujah"
    />
  </styled.div>
  <Button
    onClick={[Function]}
  >
    view more albums
  </Button>
</div>
`;
